web/src/search_suggestion.ts

```diff
@@ -588,9 +632,9 @@ function get_has_filter_suggestions(last, terms) {
     return get_special_filter_suggestions(last, terms, suggestions);
 }
 
-function get_sent_by_me_suggestions(last, terms) {
+function get_sent_by_me_suggestions(last: NarrowTerm, terms: NarrowTerm[]): Suggestion[] {
     const last_string = Filter.unparse([last]).toLowerCase();
-    const negated = last.negated || (last.operator === "search" && last.operand[0] === "-");
+    const negated = (last.operator === "search" && last.operand.startsWith("-")) || last.negated;
```

Comment:
Why did you move `last.negated` to the end?

Reply:
[Reason.](https://github.com/zulip/zulip/pull/29251#discussion_r1522397515)

Reply:
(see above)

Reply:
Done